### PR TITLE
bugfix: Remove conflicting protobuf dependency

### PR DESCRIPTION
**Pull request by doosuu**
## Describe your changes

Removes a conflicting protobuf dependency which lets build requirements and tool requirements collide.

## Checklist - Manual tasks

* [x] Examples are executing successfully
* [ ] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [ ] Created/updated integration tests.
* [x] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully
* [ ] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas)

### CODE REVIEW

File: test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+find_package(vehicle-app-sdk QUIET)
+
+if(NOT vehicle-app-sdk_FOUND)
+    message(FATAL_ERROR "This project can only be built with a 'conan create' command in the parent directory!")
+endif()
+
+add_executable(${PROJECT_NAME} main.cpp)
+target_link_libraries(${PROJECT_NAME} 
+    vehicle-app-sdk::vehicle-app-sdk
+)
```
Comment (dennismeister93) [inline test_package/CMakeLists.txt:28]: no new line

File: test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+    
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        pass
```
Comment (dennismeister93) [inline test_package/conanfile.py:41]: no new line

**Review by dennismeister93 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -30,35 +30,44 @@ on:
...
-
-      - name: Check if conan package is installed
+      - name: Create conan package
         run: |
-          RECIPE_EXISTS=$(conan search vehicle-app-sdk | grep "Existing package recipes:")
-
-          if [ "$RECIPE_EXISTS" == "" ]; then
-            exit -1
-          fi
+          conan create --build=missing -pr:b ./.conan/profiles/linux_x86_64_release -pr:h ./.conan/profiles/${{ matrix.host }}_release . ci/testing
```
Comment (MP91) [inline .github/workflows/ci.yml:56]: there is no folder ci/testing, is that maybe an old state?
```suggestion
          conan create --build=missing -pr:b ./.conan/profiles/linux_x86_64_release -pr:h ./.conan/profiles/${{ matrix.host }}_release .
```

File: test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2022-2024 Contributors to the Eclipse Foundation
```
Comment (MP91) [inline test_package/conanfile.py:1]:
```suggestion
# Copyright (c) 2024 Contributors to the Eclipse Foundation
```

File: test_package/CMakeLists.txt
Comment (MP91) [inline test_package/CMakeLists.txt:1]: what shall this test do?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -30,35 +30,44 @@ on:
...
       - name: install prerequisites
         run: |
           sudo apt-get update && sudo apt-get install -y ninja-build clang-format-14 clang-tidy-14 cppcheck
           pip3 install -r requirements.txt
 
-      - name: Export conan package
-        run: |
-          conan export .
-
-      - name: Check if conan package is installed
```
Comment (MP91) [inline .github/workflows/ci.yml:47]: why did you remove this check?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (doosuu) [reply .github/workflows/ci.yml:56]: this is the conan channel, no folder

File: .github/workflows/ci.yml (hunk 5 above)
Comment (doosuu) [reply .github/workflows/ci.yml:47]: conan create does an export to local cache and a package test, hence a conan find is not necessary

File: test_package/CMakeLists.txt
Comment (doosuu) [reply test_package/CMakeLists.txt:1]: Its mandatory to have this method, but compilation, linking and stuff is done regardless of the test method.

Comment (doosuu) [reply test_package/CMakeLists.txt:1]: https://docs.conan.io/2.0/reference/conanfile/methods/test.html we have no binaries to verify the package

File: test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+    
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        pass
```
Comment (MP91) [inline test_package/conanfile.py:41]: shouldn't we at least try to run the app once? to really test if the import is working?

Comment (doosuu) [reply test_package/conanfile.py]: Added a test for this.

**Review by MP91 — DISMISSED** [review]
lgtm

File: test_package/CMakeLists.txt (hunk 1 above)
Comment (doosuu) [reply test_package/CMakeLists.txt:28]: added

**Review by dennismeister93 — APPROVED** [review]
lgtm
